library/src/main/java/com/novoda/downloadmanager/DownloadFileStatus.java

```diff
@@ -57,7 +58,7 @@ void isMarkedAsPaused() {
     }
 
     boolean isMarkedAsError() {
-        return status == Status.ERROR;
+        return status == Status.ERROR && downloadError != null;
```

Comment:
What is the use case for if the Status is `ERROR` but there is no `downloadError`? 
You could reach this case but passing in  `Status.ERROR` in the constructor.

Reply:
Hmmm yes in theory this could happen. The point is that `DownloadFileStatus` shouldn't be created with an `ErrorStatus` that's why the `Error` was removed from the constructor signature. Unfortunately I'm not sure how to get around this without making everything **immutable** but by doing that we will end up with memory churn associated with the update callbacks.

Reply:
Yeah thats fair. I guess it would have to be a redesign or something sort, that is out of scope of fixing this.  
Definitely wouldn't block the merging of this PR on this 👍